main.js

```diff
@@ -1,158 +1,175 @@
-
-  // macOS apps generally continue running even without any windows open. 
-  // Activating the app when no windows are available should open a new one.
-  app.on('activate', () => {
-    if (BrowserWindow.getAllWindows().length === 0) renderPasswordWindow()
-  })
-});
-
-  window.setMinimumSize(200, 200);
-
-  window.loadFile("index.html");
 
-  //add event for hide and close app
-  window.on("close", (event) => {
+  // Handle close event for the password window
+  welcomeView.on("close", (event) => {
     if (!app.isQuitting) {
       event.preventDefault();
-      window.hide();
+      welcomeView.hide(); // Hide the welcome window instead of closing
     }
   });
+};
+
+app.whenReady().then(() => {
+  renderPasswordWindow();
 
-  /*use template images for a menu bar (Tray) icon, so it can adapt to both light and dark menu bars.
-  On platforms that support high pixel density displays (such as Apple Retina), we can append @2x after image's base
-  filename to mark it as a 2x scale high resolution image. */
+  // Create Tray icon
   const iconPath = path.join(__dirname, "icons", "frogIconTemplate.png");
-  const icon = nativeImage.createFromPath(iconPath);
-  icon.resize({ width: 16, height: 16 });
-  let tray = new Tray(icon);
-
-  tray.on('click', () => {
-    window.isVisible() ? window.hide() : window.show();
-    tray.closeContextMenu();
-  })
-  tray.on('right-click', () => {
-    tray.popUpContextMenu(contextMenu)
-  })
-
-  //Add cont menu for click on Tray bar icon
+  const icon = nativeImage.createFromPath(iconPath).resize({ width: 16, height: 16 });
+  
+  tray = new Tray(icon);
+
+  tray.on("click", () => {
+    const currentView = mainView || welcomeView;
+    if (currentView && !currentView.isDestroyed()) {
+      if (currentView.isVisible()) {
+        //hide if window open
+        currentView.hide(); 
+      } else {
+        //show if window hide
+        currentView.show();
+        currentView.focus();
+      }
+    }
+  });
+  
   const contextMenu = Menu.buildFromTemplate([
     {
       label: "Close Frog-app",
       type: "normal",
       click: () => {
-        app.exit(0);
+        app.isQuitting = true; // for close app in cotext menu
+        if (mainView) {
+          mainView.close();
+        }
+        app.quit();
       },
     },
   ]);
 
-  //add Workaround for Linux for context menu
-  if (process.platform === "linux") {
-    tray.setContextMenu(contextMenu);
-  }
-  
+  tray.setContextMenu(contextMenu);
   tray.setToolTip("Frog-app.");
-});
 
-// macOS apps generally continue running even without any windows open.
-// Activating the app when no windows are available should open a new one.
-app.on("activate", () => {
-  if (BrowserWindow.getAllWindows().length === 0) createWindow();
+  app.on('activate', () => {
+    if (BrowserWindow.getAllWindows().length === 0) renderPasswordWindow();
+  });
+
+  // Path for dir with txt file
+  const userFilesDir = path.join(__dirname, "User files");
```

Comment:
First of all, i think we have to out this path in env variable (i.e. file environment)
`USER_FILE_PATH=/files/text.txt`
Obviously that path is relative

Reply:
and also, I think this is a strange name for directory, never use with space